tests/integ/s3_utils.py

```diff
@@ -27,3 +29,17 @@ def assert_s3_files_exist(sagemaker_session, s3_url, files):
         found = [x["Key"] for x in contents if x["Key"].endswith(f)]
         if not found:
             raise ValueError("File {} is not found under {}".format(f, s3_url))
+
+
+def assert_s3_file_patterns_exist(sagemaker_session, s3_url, file_patterns):
+    parsed_url = urlparse(s3_url)
+    region = sagemaker_session.boto_region_name
+    s3 = boto3.client("s3", region_name=region)
+    contents = s3.list_objects_v2(Bucket=parsed_url.netloc, Prefix=parsed_url.path.lstrip("/"))[
+        "Contents"
+    ]
+    for pattern in file_patterns:
+        search_pattern = re.compile(pattern)
+        found = [x["Key"] for x in contents if search_pattern.search(x["Key"])]
+        if not found:
+            raise ValueError("File {} is not found under {}".format(f, s3_url))
```

Comment:
Bug here caused by the variable `f`, which needs to be replaced with the loop variable `pattern`.

Will wait for all CI tests to return results before submitting next commit that fixes this.